### PR TITLE
feat(event search): Add support for quoted strings

### PR DESCRIPTION
**Pull request by macqueen**

### CODE REVIEW

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -18,7 +18,9 @@
 time_filter     = "timestamp" operator date_formats
 
 search_key      = ~r"[a-z]*\.?[a-z]*"
-search_value    = ~r"\S*"
+search_value    = quoted_value / value
+value           = ~r"\S*"
+quoted_value    = ~r"\"(.+?)\""
```
[the remark this answers is not in the document]
Comment (markstory) [reply src/sentry/api/event_search.py]: Does `.+?` grab newlines or tabs, and do we want it to?

File: tests/sentry/api/test_event_search.py [hunk 2]
```diff
@@ -96,6 +96,15 @@ def test_parse_search_query_timestamp(self):
             ),
         ]
 
+    def test_parse_search_query_quoted_val(self):
+        assert parse_search_query('release:"a release"') == [
```
Comment (markstory) [inline tests/sentry/api/test_event_search.py]: Some other data I would try:

* `release:"a\nrelease"`
* `release:"a\trelease"`
* `release:"a\"\"release"`
* `release:"a release\"` (intentionally make a mistake on escaping)
* `release:\"a release "` (intentionally make a mistake on escaping)

Comment (macqueen) [reply tests/sentry/api/test_event_search.py]: added

**Review by macqueen — COMMENTED** [review]
@mattrobenolt @markstory made a few small changes and added some tests to at least demonstrate current behavior. lmk if some of that behavior seems off -- for some of the weird cases, I'm not totally sure how we _want_ things to behave

**Review by markstory — APPROVED** [review]

**Pull request merged**